jena-geosparql/src/main/java/org/locationtech/jts/index/strtree/STRtreeSerializer.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.locationtech.jts.index.strtree;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.locationtech.jts.geom.Envelope;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.Serializer;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+
+/* This file is an adapted copy of org.locationtech.jts.index.strtree.IndexSerde. */
```

Comment:
Is this via sedona?
What's the license of the code?

I think it is better to put the class in a org.apache.jena package. Jena is a library so maybe classed from the locationtech package are on the classpath another route.

As it is a single, modified class, maybe like HttpSC, which is from Jetty and copied into Jena.

Reply:
This class for serializing JTS 'STRtree' structures is from Sedona but it requires access to non-public parts of `STRtree`. So unfortunately it must reside in the `org.locationtech.jts.index.strtree` package.

Reply:
Could you put that it comes from Sedona in the comment please so that it is clear is not derived from JTS code?
And a note about why it is in the package it is in would be good.

Whether having classes in the same package spread across jars will continue working is anyone's guess, although shading crushes the distinction.